components/script/webdriver_handlers.rs

```diff
@@ -77,12 +78,27 @@ fn find_node_by_unique_id(
     pipeline: PipelineId,
     node_id: String,
 ) -> Result<DomRoot<Node>, ErrorStatus> {
-    match documents.find_document(pipeline).and_then(|document| {
-        document
-            .upcast::<Node>()
-            .traverse_preorder(ShadowIncluding::Yes)
-            .find(|node| node.unique_id() == node_id)
-    }) {
+    match documents.find_document(pipeline) {
+        Some(doc) => find_node_by_unique_id_in_document(&doc, node_id),
+        None => {
+            if ScriptThread::has_node_id(&node_id) {
+                Err(ErrorStatus::StaleElementReference)
+            } else {
+                Err(ErrorStatus::NoSuchElement)
```

Comment:
https://w3c.github.io/webdriver/#dfn-no-such-window

I think the error code should be `NoSuchWindow`
```suggestion
                Err(ErrorStatus::NoSuchWindow)
```

Reply:
I see. I got it wrong because I saw following a lot.
> "If session's [current browsing context](https://w3c.github.io/webdriver/#dfn-current-browsing-context) is [no longer open](https://w3c.github.io/webdriver/#dfn-no-longer-open), return [error](https://w3c.github.io/webdriver/#dfn-error) with [error code](https://w3c.github.io/webdriver/#dfn-error-code) [no such window](https://w3c.github.io/webdriver/#dfn-no-such-window)."

But what we really want, is to return `NoSuchWindow` to reply channel in the following?
https://github.com/servo/servo/blob/cbc363bedd5f48e08b4311b00421256d38bf488e/components/constellation/constellation.rs#L4506-L4511

Reply:
That's one possibility. It's more complicated because there isn't a single common channel, because it's stored inside variants of the actual command enum.